internal/common/debug/types.go

```diff
@@ -43,6 +43,32 @@ type (
 	// stats on the Worker for debugging purposes.
 	// Deprecated: in development and very likely to change
 	WorkerStats struct {
-		PollerTracker PollerTracker
+		PollerTracker   PollerTracker
+		ActivityTracker ActivityTracker
+	}
+
+	// ActivityInfo contains details on the executing activity
+	// Deprecated: in development and very likely to change
+	ActivityInfo struct {
+		WorkflowID   string
```

Comment:
We cannot have workflowIDs/RunIDs as a granularity because this will explode in memory.
Maps only grow and will never be downscaled. I remember that was my proposal, but I was wrong here :(

Reply:
🥲